enacts/wat_bal/maproom_monit.py

```diff
@@ -381,37 +381,34 @@ def wat_bal_ts(
     planting_year=None,
     time_coord="T",
 ):
-    water_balance_outputs = wat_bal(
-    precip,
-    et,
-    taw,
-    planting_day,
-    planting_month,
-    kc_init_length,
-    kc_veg_length,
-    kc_mid_length,
-    kc_late_length,
-    kc_init,
-    kc_veg,
-    kc_mid,
-    kc_late,
-    kc_end,
-    planting_year=planting_year,
-    time_coord=time_coord,
-)
-    for wbo in water_balance_outputs:
-        if map_choice == "paw" and wbo.name == "sm":
-            ts = 100 * wbo / taw
-        elif map_choice == "water_excess" and wbo.name == "sm":
-            ts = xr.DataArray(
-                np.isclose(wbo, taw).cumsum(),
-                dims="T",
-                coords={"T": wbo["T"]},
-            )
-        elif map_choice == "peff":
-            ts = precip_effective
-        elif (wbo.name == map_choice):
-            ts = wbo
+    try:
+        water_balance_outputs = wat_bal(
+            precip, et, taw,
+            planting_day, planting_month,
+            kc_init_length,
+            kc_veg_length,
+            kc_mid_length,
+            kc_late_length,
+            kc_init, kc_veg, kc_mid, kc_late, kc_end,
+            planting_year=planting_year,
+            time_coord=time_coord,
+        )
+        for wbo in water_balance_outputs:
+            if map_choice == "paw" and wbo.name == "sm":
+                ts = 100 * wbo / taw
+            elif map_choice == "water_excess" and wbo.name == "sm":
+                ts = xr.DataArray(
+                    np.isclose(wbo, taw).cumsum(),
+                    dims="T",
+                    coords={"T": wbo["T"]},
+                )
+            elif map_choice == "peff":
+                ts = precip_effective
+            elif (wbo.name == map_choice):
+                ts = wbo
+    except TypeError:
+        #Later tested to return error image rather than broken one
+        ts = None
```

Comment:
Is there any information available to explain why it failed? (If not, this is fine.)

Reply:
Before running wat_bal_ts, the location (X/Y) has already been applied and tested. So here, it's mostly to fail graciously if something else goes wrong, in particular poor choices of callback Input. And what comes right after calling each of the 2 wat_bal_ts call is:

if (ts is None):
        return pingrid.error_fig(
            error_msg="Please ensure all input boxes are filled for the calculation to run."
        )

Reply:
I am going to resolve this conversation so I can merge. We can always get back to this in another PR.